pkg/services/baremetal/remediation/remediation.go

```diff
@@ -51,56 +50,41 @@ func NewService(scope *scope.BareMetalRemediationScope) *Service {
 }
 
 // Reconcile implements reconcilement of HetznerBareMetalRemediations.
-func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err error) {
-	// try to get information about host from bare metal machine annotations
-	key, err := objectKeyFromAnnotations(s.scope.BareMetalMachine.ObjectMeta.GetAnnotations())
+func (s *Service) Reconcile(ctx context.Context) (reconcile.Result, error) {
+	host, err := host.GetAssociatedHost(ctx, s.scope.Client, s.scope.BareMetalMachine)
 	if err != nil {
-		if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
-			err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
-			record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-			return res, err
-		}
-		record.Event(s.scope.BareMetalRemediation, "ExitRemediation", "exit remediation because bare metal machine has no host annotation")
-		return res, nil
-	}
-
-	// get host
-	var host infrav1.HetznerBareMetalHost
-	if err := s.scope.Client.Get(ctx, key, &host); err != nil {
 		if apierrors.IsNotFound(err) {
-			if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
-				err = fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
-				record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-				return res, err
-			}
-			record.Event(s.scope.BareMetalRemediation, "ExitRemediation", "host not found - exit remediation")
-			return res, nil
+			return s.setOwnerRemediatedConditionToFailed(ctx,
+				"exit remediation because host not found")
 		}
-		err := fmt.Errorf("failed to find the unhealthy host: %w", err)
+
+		// retry
+		err := fmt.Errorf("failed to find the unhealthy host (will retry): %w", err)
 		record.Warn(s.scope.BareMetalRemediation, "FailedToFindHost", err.Error())
-		return res, err
+		return reconcile.Result{}, err
 	}
 
-	// if host is not provisioned or in maintenance mode, then we do not try to reboot server
-	if host.Spec.Status.ProvisioningState != infrav1.StateProvisioned ||
-		host.Spec.MaintenanceMode != nil && *host.Spec.MaintenanceMode {
-		if err := s.setOwnerRemediatedConditionNew(ctx); err != nil {
-			err := fmt.Errorf("failed to set remediated condition on capi machine: %w", err)
-			record.Warn(s.scope.BareMetalRemediation, "FailedSettingConditionOnMachine", err.Error())
-			return res, err
-		}
-		record.Eventf(
-			s.scope.BareMetalRemediation,
-			"ExitRemediation",
-			"exit remediation because host is not provisioned or in maintenance mode. Provisioning state: %s. Maintenance mode: %v",
-			host.Spec.Status.ProvisioningState, host.Spec.MaintenanceMode != nil && *host.Spec.MaintenanceMode,
-		)
-		return res, nil
+	if host == nil {
```

Comment:
can this happen? We handle the not found error... at least it wouldn't make sense

Reply:
I would like to change it. Before, it was not a library function, so it was clear from the inline code. Right now, with having this in a separate package, that looks strange.